### PR TITLE
Transition node to typescript in zenith

### PR DESCRIPTION
**Pull request by TomChv**
Replace #6373 with a simpler set of changes that only concerns our codebase and Zenith documentation.

#### List of updates:
- Our GitHub Actions (/cc @gerhard) 
- Our release doc (/cc @jedevc) 
- `sdk/nodejs` -> `sdk/typescript`
- Mage `sdk:nodejs` -> `sdk:typescript`
- Our generator
- I created a label `sdk/typescript` on GH

I also deleted the reference directory in `current_docs`, since these are generated files.

This PR should be way easier to review than the previous one since most changes are path related.

/cc @vikram-dagger @helderco 

Related to DEV-3212

### CODE REVIEW

**Comment by vikram-dagger** [comment]
Thanks for clearing out the auto-gen files. Speaking only for the docs, the changes seem fine. Will leave it to others to comment on the codebase changes.

File: sdk/typescript/README.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
-# Dagger NodeJS SDK
+# Dagger TypeScript SDK
 
 A client package for running [Dagger](https://dagger.io/) pipelines.
 
-## What is the Dagger NodeJS SDK?
+## What is the Dagger TypeScript SDK?
 
-The Dagger NodeJS SDK contains everything you need to develop CI/CD pipelines in Typescript of Javascript, and run them on any OCI-compatible container runtime.
+The Dagger TypeScript SDK contains everything you need to develop CI/CD pipelines in TypeScript of Javascript, and run them on any OCI-compatible container runtime.
```
Comment (jedevc) [inline sdk/typescript/README.md:7]:
```suggestion
The Dagger TypeScript SDK contains everything you need to develop CI/CD pipelines in TypeScript or Javascript, and run them on any OCI-compatible container runtime.
```

Comment (TomChv) [reply sdk/typescript/README.md]: Done in https://github.com/dagger/dagger/pull/6381/commits/5a635fc29b6b40c36fbe42fd315a12b29dd0a4be

File: RELEASING.md [hunk 2]
```diff
@@ -303,26 +303,26 @@ gh release create "sdk/python/${PYTHON_SDK_VERSION:?must be set}" \
...
 
 - [ ] Tag & publish:
 
 ```console
-cd sdk/nodejs && export NODEJS_SDK_VERSION=$(changie latest) && cd ../..
-git tag "sdk/nodejs/${NODEJS_SDK_VERSION:?must be set}" "${SDK_GIT_SHA:?must be set}"
-git push "${DAGGER_REPO_REMOTE:?must be set}" sdk/nodejs/${NODEJS_SDK_VERSION}
+cd sdk/typescript && export TYPESCRIPT_SDK_VERSION=$(changie latest) && cd ../..
+git tag "sdk/typescript/${TYPESCRIPT_SDK_VERSION:?must be set}" "${SDK_GIT_SHA:?must be set}"
+git push "${DAGGER_REPO_REMOTE:?must be set}" sdk/typescript/${NODEJS_SDK_VERSION}
```
[the remark this answers is not in the document]
Comment (TomChv) [reply RELEASING.md]: Changes in https://github.com/dagger/dagger/pull/6381/commits/c9bd9727c72287d557768a42984aaae081fd9218

**Review by jedevc — APPROVED** [review]

**Comment by jedevc** [comment]
LGTM once tests pass :heart:

**Pull request merged**